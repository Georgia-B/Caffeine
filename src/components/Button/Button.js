import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.module.css';

const Button = ({ text, type, onClick }) => {
    return (
        <button type={type} className={styles.button} onClick={onClick}>
            {text}
        </button>
    );
};

Button.propTypes = {
    text: PropTypes.string,
    type: PropTypes.string,
    onClick: PropTypes.func,
};

export default Button;
