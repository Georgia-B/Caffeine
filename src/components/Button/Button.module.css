.button {
    padding: 10px 15px;
    margin-top: 15px;
    background-color: #ffce62;
    border-style: none;
    border: 0;
    border-radius: 4px;
    font-size: 16px;
    font-family: inherit;
    min-width: 120px;
    cursor: pointer;
    outline: none;
}
    
.button:hover {
    background-image: linear-gradient(#ffce62,#f3c258,#e7b750);
}