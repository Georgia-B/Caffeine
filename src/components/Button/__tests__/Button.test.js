import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Button from '../Button';

describe('Components | <Button />', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Button
            text="Click me"
            type="submit"
            onClick={() => true}
        />);

        expect(component.asFragment()).toMatchSnapshot();
    });
});
