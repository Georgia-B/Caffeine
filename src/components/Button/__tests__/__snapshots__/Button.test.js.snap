// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <Button /> should render correctly 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="submit"
  >
    Click me
  </button>
</DocumentFragment>
`;
